app/src/main/java/org/oppia/app/options/StoryTextSizeFragmentPresenter.kt

```diff
@@ -0,0 +1,131 @@
+        binding.storyTextSizeSeekBar.progress = 0
+        binding.previewTextview.textSize = getStoryTextSizeInFloat(StoryTextSize.SMALL_TEXT_SIZE)
+      }
+      getStoryTextSize(StoryTextSize.MEDIUM_TEXT_SIZE) -> {
+        binding.storyTextSizeSeekBar.progress = 5
+        binding.previewTextview.textSize = getStoryTextSizeInFloat(StoryTextSize.MEDIUM_TEXT_SIZE)
+      }
+      getStoryTextSize(StoryTextSize.LARGE_TEXT_SIZE) -> {
+        binding.storyTextSizeSeekBar.progress = 10
+        binding.previewTextview.textSize = getStoryTextSizeInFloat(StoryTextSize.LARGE_TEXT_SIZE)
+      }
+      getStoryTextSize(StoryTextSize.EXTRA_LARGE_TEXT_SIZE) -> {
+        binding.storyTextSizeSeekBar.progress = 15
+        binding.previewTextview.textSize =
+          getStoryTextSizeInFloat(StoryTextSize.EXTRA_LARGE_TEXT_SIZE)
+      }
+    }
+
+    binding.storyTextSizeSeekBar.max = 15
+
+    binding.storyTextSizeSeekBar.setOnSeekBarChangeListener(
+      object :
+        SeekBar.OnSeekBarChangeListener {
+        override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
+          var progressValue = progress
+          progressValue /= 5
+          progressValue *= 5
+
+          when (progressValue) {
+            0 -> {
+              fontSize = getStoryTextSize(StoryTextSize.SMALL_TEXT_SIZE)
+              binding.previewTextview.textSize =
+                getStoryTextSizeInFloat(StoryTextSize.SMALL_TEXT_SIZE)
+            }
+            5 -> {
+              fontSize = getStoryTextSize(StoryTextSize.MEDIUM_TEXT_SIZE)
+              binding.previewTextview.textSize =
+                getStoryTextSizeInFloat(StoryTextSize.MEDIUM_TEXT_SIZE)
+            }
+            10 -> {
+              fontSize = getStoryTextSize(StoryTextSize.LARGE_TEXT_SIZE)
+              binding.previewTextview.textSize =
+                getStoryTextSizeInFloat(StoryTextSize.LARGE_TEXT_SIZE)
+            }
+            else -> {
+              fontSize = getStoryTextSize(StoryTextSize.EXTRA_LARGE_TEXT_SIZE)
+              binding.previewTextview.textSize =
+                getStoryTextSizeInFloat(StoryTextSize.EXTRA_LARGE_TEXT_SIZE)
+            }
+          }
+          seekBar.progress = progressValue
+          updateTextSize(fontSize)
+        }
+
+        override fun onStartTrackingTouch(seekBar: SeekBar) {
+        }
+
+        override fun onStopTrackingTouch(seekBar: SeekBar) {
+        }
+      })
+
+    return binding.root
+  }
+
+  fun updateTextSize(textSize: String) {
+    // The first branch of (when) will be used in the case of multipane
+    when (val parentActivity = fragment.activity) {
+      is OptionsActivity -> parentActivity.optionActivityPresenter.updateStoryTextSize(textSize)
+      is StoryTextSizeActivity ->
+        parentActivity.storyTextSizeActivityPresenter.setSelectedStoryTextSize(textSize)
+    }
+  }
+
+  fun getStoryTextSizeInFloat(storyTextSize: StoryTextSize): Float {
```

Comment:
Please add a TODO for this function improvement.
`// TODO(#1584): Update this function to use multiplier instead of direct font size.`

Reply:
Done